packages/ui-library/src/components/select/index.tsx

```diff
@@ -1,4 +1,123 @@
-import type { SelectProps } from '@mui/material/Select';
-import Select, { SelectChangeEvent } from '@mui/material/Select';
+import React, { useCallback, useMemo, useRef, useState } from 'react';
+import { MuiSelect, ListSubheader, TextField, MenuItem, MuiSelectChangeEvent, InputAdornment, Divider } from '..';
+import { SearchIcon } from '../../icons';
+import { defineMessage, useIntl } from 'react-intl';
 
-export { Select, type SelectProps, SelectChangeEvent };
+interface DisabledSearchProps {
+  disabledSearch: true;
+}
+
+interface EnabledSearchProps<T> {
+  disabledSearch: false;
+  searchFunction: (data: T, searchTerm: string) => boolean;
+}
+
+type SearchProps<T> = DisabledSearchProps | EnabledSearchProps<T>;
+
+interface BaseSelectProps<T extends { key: string | number }> {
+  disabledSearch?: boolean;
+  placeholder?: string;
+  options: T[];
+  onChange: (selectedOption: T) => void;
+  RenderItem: React.ComponentType<{ item: T }>;
+  selectedItem?: T;
+  id?: string;
+  searchFunction?: (data: T, searchTerm: string) => boolean;
+}
+
+type SelectProps<T extends { key: string | number }> = BaseSelectProps<T> & SearchProps<T>;
+
+function Select<T extends { key: string | number }>({
+  id,
+  placeholder,
+  RenderItem,
+  options,
+  onChange,
+  selectedItem,
+  disabledSearch = false,
+  searchFunction,
+}: SelectProps<T>) {
+  const [search, setSearch] = useState('');
+  const searchRef = useRef<HTMLDivElement>();
+  const intl = useIntl();
+
+  const renderedItems = useMemo(
+    () => (!disabledSearch && searchFunction ? options.filter((option) => searchFunction(option, search)) : options),
+    [search, disabledSearch, searchFunction]
+  );
+
+  const handleChangeNetwork = useCallback(
+    (evt: MuiSelectChangeEvent<number>) => {
+      const selectedOption = options.find((option) => option.key === evt.target.value);
+      if (selectedOption) {
+        setSearch('');
+        onChange(selectedOption);
+      }
+    },
+    [onChange, options]
+  );
+
+  const handleOnClose = useCallback(() => setSearch(''), []);
+
+  return (
+    <MuiSelect
```

Comment:
We need to specify icon color (in light mode is not visible): 
<img width="450" alt="Screenshot 2024-01-17 at 16 47 26" src="https://github.com/Mean-Finance/dca-fe/assets/144699868/d2cbe78c-666a-4e7c-84e5-5e4407d16462">

Reply:
Oh good catch